File: core/data/src/main/java/org/mifos/mobilewallet/core/data/fineract/api/services/TwoFactorAuthService.java
package org.mifos.mobilewallet.core.data.fineract.api.services;

import org.mifos.mobilewallet.core.data.fineract.api.ApiEndPoints;
import org.mifos.mobilewallet.core.domain.model.twofactor.AccessToken;
import org.mifos.mobilewallet.core.domain.model.twofactor.DeliveryMethod;

import java.util.List;

import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;
import rx.Observable;

/**
 * Created by ankur on 01/June/2018
 */

public interface TwoFactorAuthService {

    @GET(ApiEndPoints.TWOFACTOR)
    Observable<List<DeliveryMethod>> getDeliveryMethods();

    @POST(ApiEndPoints.TWOFACTOR)
    Observable<String> requestOTP(@Query("deliveryMethod") String deliveryMethod);

    @POST(ApiEndPoints.TWOFACTOR + "/validate")
    Observable<AccessToken> validateToken(@Query("token") String token);

}

File: core/data/src/main/java/org/mifos/mobilewallet/core/data/fineract/api/services/TwoFactorAuthService.kt
package org.mifos.mobilewallet.core.data.fineract.api.services

import org.mifos.mobilewallet.core.data.fineract.api.ApiEndPoints
import com.mifos.mobilewallet.model.domain.twofactor.AccessToken
import com.mifos.mobilewallet.model.domain.twofactor.DeliveryMethod
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query
import rx.Observable

/**
 * Created by ankur on 01/June/2018
 */
interface TwoFactorAuthService {
    @get:GET(ApiEndPoints.TWOFACTOR)
    val deliveryMethods: Observable<List<DeliveryMethod>>

    @POST(ApiEndPoints.TWOFACTOR)
    fun requestOTP(@Query("deliveryMethod") deliveryMethod: String): Observable<String>

    @POST(ApiEndPoints.TWOFACTOR + "/validate")
    fun validateToken(@Query("token") token: String): Observable<AccessToken>
}